### PR TITLE
Fix #679, Add Markdown version of OSAL Configuration Guide

### LINKED ISSUE #679: Convert doc/OSAL-Configuration-guide.doc to markdown
**Is your feature request related to a problem? Please describe.**
Doc and PDF not GitHub friendly

**Describe the solution you'd like**
Convert to markdown and remove doc/pdf

**Describe alternatives you've considered**
None

**Additional context**
None

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Describe the contribution**
Pandoc was used to convert the OSAL Configuration Guide to a markdown file. During the conversion, issues were presented such as broken tables. Tables were fixed and block codes were added where appropriate. 

Fix #679

**Additional context**
Please provide feedback on whether the markdown file is acceptable in terms of formatting. 

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

**Comment by skliper** [comment]
@astrogeco @ArielSAdamsNASA - what's the status on this?  It would help to move it forward...

**Comment by skliper** [comment]
Also delete the other versions... markdown only preferred.

**Comment by ArielSAdamsNASA** [comment]
> @astrogeco @ArielSAdamsNASA - what's the status on this? It would help to move it forward...

@skliper Waiting on @acudmore for approval/feedback. I sent him an email this morning.

**Comment by skliper** [comment]
Related to #384, the issue to update the contents.  A format check here should be sufficient (and will simplify fixing #384).

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 1]
```diff
@@ -0,0 +1,798 @@
+[National Aeronautics and Space Administration]{.smallcaps}
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:1]: Remove front matter and have the first section be the Introduction

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 2]
```diff
@@ -0,0 +1,798 @@
...
+
+**[Approvals:]{.ul}**
+
+![][4]
+
+Acknowledgements
+
+Revision History
+
+| Revision Number |  Release Date | Changes to Prior Revision  | Approval |
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:27]: Maybe move these to an appendix of sorts? I'm not sure we need this when the document is versioned using git

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+
+[2.12.2 How to Run the OSAL on the PC / Linux Platform 21]
+
+[3 OSAL Unit Tests 22]
+
+[3.1.1 OSAL Unit Test Configuration for the PC / Linux Platform 22]
+
+[3.1.2 How to Run the OSAL Unit Tests on the PC / Linux Platform 22]
+
+Introduction
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:140]: Use markdown "#" syntax for all headings

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 4]
```diff
@@ -0,0 +1,798 @@
...
+|OS_UTILITYTASK_STACK_SIZE|The size of the stack for the utility task.|
+|OS_UTILITYTASK_PRIORITY|The priority of the utility task.|
+|OSAL_SOCKET_QUEUE |If this is defined, the posix port will use the socket implementation for message queues, rather than the POSIX message queue implementation.|
+|OS_MAX_MODULES |Used for defining the maximum number of loadable modules that the OS AL can keep track of. This is used for the new Module Load and Symbol API.|
+|OS_MAX_SYM_LEN|Used for setting the maximum length of a symbol name in the symbol API.|
+|OS_MAX_TIMERS|Used for defining the maximum number of timers in the OSAL.|
+
+
+Setting up "classic" build
+--------------------------
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:287]: Use Markdown syntax and get rid of horizontal rule "-----"

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 5]
```diff
@@ -0,0 +1,798 @@
...
+|---|---|---|
+|   OS|rtems||
+|  BSP|rtems-mcf5235||
+
+
+### File System Mappings on the Axiom M5235 BCC / RTEMS 4.10
+
+The RTEMS port of the OSAL has a one to one file system mapping. The OSAL RAM disk will format an RTEMS NVRAM disk with the RFS file system.The file system mappings are defined in the bsp_voltab.c file in the src/bsp/mcf5235-rtems/src directory:
+
+**OSAL File System Mappings**
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:615]: Make this a fourth-level heading "####"

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 6]
```diff
@@ -0,0 +1,798 @@
...
+|  /ramdev1 -- /ramdev5|Unused|N/A|N/A|Unused table entries for applications to create new RAM disks. RTEMS does not currently have support for creating new RAM disks.|
+|  /ssedev0 - /ssrdev2|File System Mapped (FS_BASED)|N/A|N/A|Unused table entries for applications to map Hard Disk device directories to "pseudo" SSR file systems.|
+
+### How to run the OSAL Applications on the SPARC SIS Simulator with RTEMS 4.10
+
+When the example application and test programs are all built as static executables for the SIS Simulator built into the sparc-rtems4.10-gdb executable.
+
+To run an example or test, simply do the following:
+
+**\$ sparc-rtems4.10-gdb tasking-example.nxe**
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:663]: Use a code block for this

File: doc/OSAL-Configuration-Guide-Markdown.md [hunk 7]
```diff
@@ -0,0 +1,798 @@
...
+**
+
+PC / Linux Platform
+-------------------
+
+The OSAL can run on linux distributions. Testing is done with CentOS 6.5 **32 bit** and Ubuntu 13.10 64 bit. Newer versions of the Linux 2.6 kernel have POSIX message queues, which can be used for the OSAL Queue implementation. If the POSIX message queues are not available, then the OSAL Queues can use UDP sockets. (see the OS_SOCKET_QUEUE configuration parameter). In general, the older versions of linux (2.4 kernel) are not supported, but most modern Linux releases such as Ubuntu 12.10 and later, and Cent OS/Redhat Enterprise Linux 5 and later should work. The OSAL has also been used on the Raspberry Pi computer with the Raspbian Debian based linux. The OSAL is primarily run on 32 bit linux distributions, but it should compile and run as a 32 bit application on 64 bit linux.
+
+### OSAL Configuration for the PC / Linux Platform
+
+**osal-config.mak Settings**
```
Comment (astrogeco) [inline doc/OSAL-Configuration-Guide-Markdown.md:683]: Use Fourth-level headings "####"

**Review by astrogeco — CHANGES_REQUESTED** [review]
Mainly format cleanups. It might be worthwhile to do a find-and-replace on bold and italic syntax "***" which is sometimes used for subheadings in this document. Similarly horizontal rules "===" are also being used in a subheading fashion. 

It might be worth passing this through a markdown linter like remark or prettier; there are Atom and VSCode plugins for to help with that.

**Review by astrogeco — CHANGES_REQUESTED** [review]
Remove "markdown" from the file name and use [`git rm` ](https://www.atlassian.com/git/tutorials/undoing-changes/git-rm) to remove the pdf and doc versions of the guide.

File: doc/OSAL-Configuration-Guide.md [hunk 8]
```diff
@@ -0,0 +1,873 @@
+# Introduction
```
Comment (ArielSAdamsNASA) [inline doc/OSAL-Configuration-Guide.md:1]: Removed front matter so the document begins with the Introduction.

File: doc/OSAL-Configuration-Guide.md [hunk 9]
```diff
@@ -0,0 +1,873 @@
...
+ | **Directory**|**Description**|
+ |---|---|
+ |  build|The top level OSAL build directory.|
+ |  build/unit-tests|The top level unit test build directory.|
+ |  build/unit-tests/oscore-test|Contains the test binary, log, and gcov files for the OSAL core tests|  |  build/unit-tests/osfile-test|Contains the test binary, log, and gcov files for the OSAL file API tests|  
+ | build/unit-tests/osfilesys-test|Contains the test binary, log, and gcov files for the OSAL file system tests|  
+ | build/unit-tests/osloader-test|Contains the test binary, log, and gcov files for the OSAL loader tests|  
+ |build/unit-tests/ostimer-test|Contains the test binary, log, and gcov files for the OSAL timer tests| 
+
+# Revision History
```
Comment (ArielSAdamsNASA) [inline doc/OSAL-Configuration-Guide.md:861]: Placed Revision History at the bottom of the document for now.

**Comment by ArielSAdamsNASA** [comment]
I removed the pdf and doc versions of the guide and updated the markdown version.

For the markdown version:
- Removed the front matter, so the Introduction is the first section 
- Updated headers to use the markdown syntax # and removed horizontal rule for headers
- Updated table headers to use fourth-level headers 
- Use a code block for the section starting with **\$ sparc-rtems4.10-gdb tasking-example.nxe**
- Moved Revision History to the bottom of the page for now

File: doc/OSAL-Configuration-Guide.md [hunk 10]
```diff
@@ -0,0 +1,873 @@
...
+|  TBD | To Be Determined |  
+
+## Glossary of Terms
+
+The following table defines the terms used throughout this document.
+These terms are identified as proper nouns and are capitalized. 
+
+| **Term**    |  **Definition**     |  
+|---|---|
+| Application  (APP)   |  A generic term for a computer program in a
```
Comment (acudmore) [inline doc/OSAL-Configuration-Guide.md:45]: This table does not look like it is rendering correctly, at least when viewed on GitHub.
This is what it looks like when I view it.
osal-markdown

**Review by acudmore — APPROVED** [review]
The conversion is good, there are a couple of table format issues ( in the glossary for example ). But this conversion and removing the old documents is the first step, I think the doc needs to be overhauled in content after this conversion. Much of the technical info is out of date.

File: doc/OSAL-Configuration-Guide.md (hunk 10 above)
Comment (ArielSAdamsNASA) [reply doc/OSAL-Configuration-Guide.md]: Updated several tables in terms of formatting. Should be all set to be marked as CCB-READY?

**Comment by astrogeco** [comment]
Looks good! I would do an interactive rebase and squash the table fix into the original md commit. Also edit the commit message to use the format "Fix #XYZ, COMMIT_DESCRIPTION"

**Comment by astrogeco** [comment]
[CCB 2020-12-16](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-December-16-2020) APPROVED

- Separate issue for content

**Review by astrogeco — APPROVED** [review]

**Pull request merged**